webhooks/views.py

```diff
@@ -45,35 +50,83 @@ class ContentFileWebhookView(BaseWebhookView):
+        if response.is_valid():
+            return Response(response.data)
+        else:
+            log.error("Invalid response data: %s", response.errors)
+            return HttpResponseBadRequest("Invalid response data")
+
+    def get_data(self, request):
+        """
+        Get data from the serializer
         """
+        serializer = ContentFileWebHookRequestSerializer(data=json.loads(request.body))
+        if not serializer.is_valid():
+            log.error("Invalid webhook data: %s", serializer.errors)
+            msg = "Invalid data"
+            raise BadRequest(msg)
+        return serializer.validated_data
+
+    def post(self, request):
         try:
-            serializer = ContentFileWebHookRequestSerializer(
-                data=json.loads(request.body)
-            )
-            if not serializer.is_valid():
-                log.error("Invalid webhook data: %s", serializer.errors)
-                return HttpResponseBadRequest("Invalid data")
-            data = serializer.validated_data
+            data = self.get_data(request)
             log.info("Received webhook data: %s", data)
-            process_content_file_request(data)
-            response = WebhookResponseSerializer(
-                data={"status": "success", "message": "Webhook received"}
-            )
-            if response.is_valid():
-                return Response(response.data)
+            process_create_content_file_request(data)
+            return self.success()
+        except json.JSONDecodeError:
+            return HttpResponseBadRequest("Invalid JSON format")
+
+
+class ContentFileDeleteWebhookView(ContentFileWebhookView):
+    """
+    Webhook handler for ContentFile DELETE requests
+    """
+
+    def post(self, request):
+        try:
+            data = self.get_data(request)
+            process_delete_content_file_request(data)
+            return self.success()
         except json.JSONDecodeError:
             return HttpResponseBadRequest("Invalid JSON format")
 
 
-def process_content_file_request(data):
+def process_create_content_file_request(data):
     """
-    Process a content file webhhok request based on the ETL source
+    Process a content file CREATE webhook request based on the ETL source
     """
     etl_source = data.get("source")
     content_path = data.get("content_path")
     if etl_source == ETLSource.canvas.name:
         log.info("Processing Canvas content file: %s", content_path)
         ingest_canvas_course.apply_async([content_path, True])
+
+
+def process_delete_content_file_request(data):
+    """
+    Process a content file DELETE webhhook request based on the ETL source
+    """
+    etl_source = data.get("source")
+    if etl_source == ETLSource.canvas.name:
+        course_id = data.get("course_id")
+        if course_id:
+            try:
+                resource = LearningResource.objects.get(
+                    readable_id__istartswith=course_id, etl_source=ETLSource.canvas.name
```

Comment:
This could potentially cause problems if the course_id is `287` and there is also a canvas course_id of `2871`.  Since the `readable_id` is `<canvas_course_id>_<course_code>`, might be best to change this slightly to include the `_`:  

```python
readable_id__istartswith=f"{course_id}_"
```